### PR TITLE
Adding config.py.example

### PR DESCRIPTION
**Pull request by sengineer0**
Ref:
https://github.com/biothings/biothings.api/issues/217
https://github.com/biothings/biothings.api/issues/218

### CODE REVIEW

**Comment by newgene** [comment]
@sengineer0 I think just a `config.py.example`  is not good enough. We should be able to store the default value within the hub, then user's `config.py` can override it or skip that parameter to use the default value.

**Comment by sengineer0** [comment]
> @sengineer0 I think just a `config.py.example` is not good enough. We should be able to store the default value within the hub, then user's `config.py` can override it or skip that parameter to use the default value.

ok got it. I will check to see how we can achieve that

**Comment by newgene** [comment]
@sengineer0 two documentation requests:

* `default_config.py` - at the top of this file, please add some doc about "what this file is, and how to define a new hub config setting?" (All future new hub settings should be defined here first, with the default value and inline docs).
* `ConfigurationWrapper` - Update the doc strings for `ConfigurationWrapper` class and `_parse_comments` method, where you added `default_conf_mod` argument. 

In general, this should be the best practice for you to document well your changes in the code.

File: config.py.example [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        desc="Define path to folder which will contain log files")
+# Usually inside DATA_ARCHIVE_ROOT
+#LOG_FOLDER = os.path.join(DATA_ARCHIVE_ROOT,'logs')
+                                                                                                                                                                                                    
+# default hub logger
+logger = ConfigurationDefault(
+        default=logging,
+        desc="Provide a default hub logger instance (use setup_default_log(name,log_folder)")
+# Usually use default setup
+#logger = setup_default_log("hub", LOG_FOLDER)
```
Comment (newgene) [inline config.py.example]: Can we remove this block from `config.py`? We can move the logic to `ConfigurationWrapper`:

* if `LOG_FOLDER` is provided, but `logger` is not defined, we will create `logger` automatically.

File: config.py.example [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+#        "uri" : "mongodb://localhost:27017",
+#        #"uri" : "mongodb://user:passwd@localhost:27017", # mongodb std URI
+#        }
+#HUB_DB_BACKEND = {
+#        "module" : "biothings.utils.es",
+#        "host" : "localhost:9200",
+#        }
+
+# cache file format ("": ascii/text uncompressed, or "gz|zip|xz"
+CACHE_FORMAT = "xz"
```
Comment (newgene) [inline config.py.example]: settings like `CACHE_FORMAT`, `BIOTHINGS_ROLE` and others below are not found in the `default_config.py`. They should be defined in `default_config.py` and removed from this `config.py.example`, right? Most hubs won't change them. There should not be any sdk-level settings here not be defined in default.

At the end of this file, you can add a commented section for any possible user-hub specific settings if needed.

File: config.py.example [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+#
+# then define the following variables to fit your needs. You can also override any
+# any other variables in this file as required. Variables defined as ValueError() exceptions
+# *must* be defined
+#
+
+import logging
+
+from biothings.utils.loggers import setup_default_log
+from biothings.utils.configuration import ConfigurationError, ConfigurationDefault, ConfigurationValue
```
Comment (newgene) [inline config.py.example]: In this `config.py.example`, we probably should not use `ConfigurationError, ConfigurationDefault, ConfigurationValue`, but using an example value (e.g. `localhost` for `DATA_SRC_SERVER`). That's easier for users to modify.

File: config.py.example (hunk 1 above)
Comment (sengineer0) [reply config.py.example]: It's updated

File: config.py.example (hunk 2 above)
Comment (sengineer0) [reply config.py.example]: It's updated

File: config.py.example (hunk 3 above)
Comment (sengineer0) [reply config.py.example]: It's updated

**Pull request merged**